x-pack/plugins/apm/common/agent_configuration/runtime_types/duration_rt.ts

```diff
@@ -6,32 +6,68 @@
 
 import * as t from 'io-ts';
 import { either } from 'fp-ts/lib/Either';
-import { amountAndUnitToObject } from '../amount_and_unit';
+import moment, { unitOfTime } from 'moment';
+import { i18n } from '@kbn/i18n';
+import { amountAndUnitToObject, AmountAndUnit } from '../amount_and_unit';
+import { getRangeType } from './get_range_type';
 
-export const DURATION_UNITS = ['ms', 's', 'm'];
+function getDuration({ amount, unit }: AmountAndUnit) {
+  return moment.duration(amount, unit as unitOfTime.Base);
+}
+
+export function getDurationRt({
+  min,
+  max,
+  units
+}: {
+  min?: string;
+  max?: string;
+  units: string[];
+}) {
+  const minAmountAndUnit = min && amountAndUnitToObject(min);
+  const maxAmountAndUnit = max && amountAndUnitToObject(max);
 
-export function getDurationRt({ min }: { min: number }) {
+  const message = i18n.translate('xpack.apm.agentConfig.duration.errorText', {
+    defaultMessage: `{rangeType, select,
+      between {Must be between {min} and {max}}
+      gt {Must be greater than {min}}
+      lt {Must be less than {max}}
+      other {Must be an integer}
+    }`,
+    values: {
+      min,
+      max,
+      rangeType: getRangeType(
+        minAmountAndUnit ? minAmountAndUnit.amount : undefined,
+        maxAmountAndUnit ? maxAmountAndUnit.amount : undefined
+      )
+    }
+  });
```

Comment:
Can you move this method to `get_range_type` and call it `getRangeTypeMessage` and re-use it for duration, bytes and integer?

Reply:
What do you think about this:
```js
export function getRangeTypeMessage(
  min?: number | string,
  max?: number | string
) {
  return i18n.translate('xpack.apm.agentConfig.range.errorText', {
    defaultMessage: `{rangeType, select,
        between {Must be between {min} and {max}}
        gt {Must be greater than {min}}
        lt {Must be less than {max}}
        other {Must be an integer}
      }`,
    values: {
      min,
      max,
      rangeType: getRangeType(
        typeof min === 'string' ? amountAndUnitToObject(min).amount : min,
        typeof max === 'string' ? amountAndUnitToObject(max).amount : max
      )
    }
  });
}
```

I min and max must be number for `integer_rt` and string for `duration_rt` and `bytes_rt`, because we must show the message like `Must be between 1ms and 1s` or `Must be between 1b and 1mb`

Reply:
Looks good 👍